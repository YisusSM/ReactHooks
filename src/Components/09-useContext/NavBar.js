import React from 'react'
import { Link } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <div className="container-fluid">
            <Link className="navbar-brand" to = "/">useContext</Link>
            <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                    <li className="nav-item">
                        <Link to="/" className="nav-link nav-item">Home</Link>
                    </li>
                    <li className="nav-item">
                        <Link to="/about" className="nav-link nav-item">About</Link>
                    </li>
                    <li className="nav-item">
                        <Link to="/login" className="nav-link nav-item">Login</Link>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
    )
}
