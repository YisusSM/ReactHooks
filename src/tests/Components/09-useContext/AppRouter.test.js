import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../Components/09-useContext/AppRouter';
import { UserContext } from '../../../Components/09-useContext/UserContext';

describe('pruebas en AppRouter', () => {

    const wrapper = mount(
        <UserContext.Provider value = {{
            id:1,
            name:'jesus'
        }}>
            <AppRouter />
        </UserContext.Provider>

    )
    test('debe mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    })

})
