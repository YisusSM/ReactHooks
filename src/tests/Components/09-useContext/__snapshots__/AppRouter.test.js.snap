// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en AppRouter debe mostrarse correctamente  1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link nav-item"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link nav-item"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link nav-item"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre />
    </div>
  </div>
</div>
`;
